src/metadata/matroska_handler.cc

```diff
@@ -146,24 +146,24 @@ void MatroskaHandler::parseMKV(const std::shared_ptr<CdsItem>& item, std::unique
     file_io_callback ebml_file(item->getLocation().c_str());
     EbmlStream ebml_stream(ebml_file);
 
-    auto el_l0 = ebml_stream.FindNextID(LIBMATROSKA_NAMESPACE::KaxSegment::ClassInfos, ~0);
-    while (el_l0 != nullptr) {
+    for (;;) {
+        auto el_l0 = ebml_stream.FindNextID(LIBMATROSKA_NAMESPACE::KaxSegment::ClassInfos, ~0);
```

Comment:
what about
```
for (
    auto el_l0 = ebml_stream.FindNextID(LIBMATROSKA_NAMESPACE::KaxSegment::ClassInfos, ~0);
    el_l0 != nullptr;
    el_l0 = ebml_stream.FindNextElement(LIBMATROSKA_NAMESPACE::KaxSegment_Context, i_upper_level, ~0, true)
)
```
Seems a lot closer to the original logic: you replaced `FindNextElement` by `FindNextID`

Reply:
Looks uglier. Might as well keep the original loop.